### PR TITLE
[DEAD-CODE] Added failing RemoveUnusedPrivateMethodRector test

### PR DESCRIPTION
**Pull request by clxmstaab**
the whole private method `notcalled` should be removed because its private and not called within the same class

https://getrector.org/demo/6831aa4c-3092-4f1a-afb7-2e25dd0f232f

closes https://github.com/rectorphp/rector/issues/5876

### CODE REVIEW

File: rules-tests/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector/Fixture/non-static-method.php.inc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\ClassMethod\RemoveUnusedPrivateMethodRector\Fixture;
+
+class DemoFile
```
Comment (staabm) [inline rules-tests/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector/Fixture/non-static-method.php.inc:5]:
```suggestion
class NonStaticMethod
```

File: rules-tests/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector/Fixture/non-static-method.php.inc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\ClassMethod\RemoveUnusedPrivateMethodRector\Fixture;
+
+class DemoFile
```
Comment (staabm) [inline rules-tests/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector/Fixture/non-static-method.php.inc:27]:
```suggestion
class NonStaticMethod
```

**Comment by clxmstaab** [comment]
hmm not sure why the error I was able to reproduce on getrector.org/demo/6831aa4c-3092-4f1a-afb7-2e25dd0f232f does not reproduce with this new test-fixture

**Comment by clxmstaab** [comment]
after fiddling a bit more locally it seems the current git version is doing the 'rectoring' as expected.

not sure why the demo sandbox on the website does not.

**Comment by staabm** [comment]
The sandbox now agrees with my unit test. It works like expected. Closing

**Comment by Bl00D4NGEL** [comment]
@clxmstaab / @staabm Guess you forgot to close the PR?

**Comment by samsonasik** [comment]
@staabm thank you for verify. I am closing it then

**Pull request closed without merging**